stdlib/builtins.pyi

```diff
@@ -327,6 +327,8 @@ class complex:
 class _FormatMapMapping(Protocol):
     def __getitem__(self, __key: str) -> Any: ...
 
+_StartEndArg = Optional[Union[int, SupportsIndex]]
```

Comment:
`SupportsIndex` includes `int`, because `int` defines `__index__` method:

```suggestion
_StartEndArg = Optional[SupportsIndex]
```

Now the type is shorter, so maybe just write `Optional[SupportsIndex]` everywhere instead of defining a new type alias.